### PR TITLE
CDPS-908: Add latestBooking to migration requests

### PR DESCRIPTION
**Pull request by davidatmoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/WithFieldHistory.kt [hunk 1]
```diff
@@ -53,7 +54,7 @@ abstract class WithFieldHistory<T : AbstractAggregateRoot<T>?> : AbstractAggrega
 
           // Set appliesTo on previous history item if not already set
           previousVersion
-            ?.takeIf { it.appliesTo == null }
+            ?.takeIf { !it.anomalous && it.appliesTo == null }
```
Comment (brightonsbox) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/jpa/WithFieldHistory.kt:57]: 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/dto/response/FieldHistoryDto.kt [hunk 2]
```diff
@@ -48,4 +48,7 @@ data class FieldHistoryDto(
 
   @Schema(description = "The source of the change, either DPS or NOMIS. Will be NOMIS if the record was either migrated or synced from NOMIS")
   val source: String? = null,
+
+  @Schema(description = "A flag to indicate that when the data was migrated from NOMIS, the booking it was taken from did not have an end date")
```
Comment (brightonsbox) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/dto/response/FieldHistoryDto.kt:52]: Maybe `, the booking it was taken from was historical but did not have an end date`

File: src/main/resources/db/migration/common/V12__add_anomalous_column_to_field_history.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+alter table field_history
+    add column anomalous boolean;
+update field_history
+set anomalous = false
+where 1 = 1;
+alter table field_history
+    alter column anomalous set not null;
+COMMENT ON COLUMN field_history.anomalous IS 'A flag to indicate that when the data was migrated from NOMIS, the booking it was taken from did not have an end date';
```
Comment (brightonsbox) [inline src/main/resources/db/migration/common/V12__add_anomalous_column_to_field_history.sql:8]: Meh.. I'd been following the capitalise keywords, lowercase names convention: https://www.sqlstyle.guide/#query-syntax

File: src/main/resources/db/migration/common/V12__add_anomalous_column_to_field_history.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+alter table field_history
+    add column anomalous boolean;
```
Comment (brightonsbox) [inline src/main/resources/db/migration/common/V12__add_anomalous_column_to_field_history.sql:2]: I think this should work:

```
ALTER TABLE field_history
ADD COLUMN anomalous BOOLEAN NOT NULL DEFAULT false
```

Then with the default of false, you could avoid specifying the `anomalous` column in tests if they are all going to be false (unless of course you want to test the anomolous behaviour in a particular test)

Comment (davidatmoj) [reply src/main/resources/db/migration/common/V12__add_anomalous_column_to_field_history.sql:2]: I didn't want to have the column default in normal operation as the API request should specify the value, so just added it this way to handle not having a blank DB when this goes in.
And yeah, I will be adding some tests to check anomalous behaves correctly.

**Review by brightonsbox — APPROVED** [review]
LGTM

**Pull request merged**